proposals/0292-package-registry-service.md

```diff
@@ -163,16 +163,9 @@ A valid package name matches the following regular expression pattern:
 > For more information,
 > see [Unicode Identifier and Pattern Syntax][UAX31].
 
-Package scopes are case-insensitive
-(for example, `mona` ≍ `MONA`).
-Package names are
-case-insensitive,
-diacritic-insensitive
-(for example, `Å` ≍ `A`), and
-width-insensitive
-(for example, `Ａ` ≍ `A`).
 Package names are compared using
-[Normalization Form Compatible Composition (NFKC)][UAX15].
+[Normalization Form Compatible Composition (NFKC)][UAX15]
+with locale-independent case folding.
```

Comment:
cc @SDGGiesbrecht @benrimmington to confirm this addresses the concerns in https://forums.swift.org/t/se-0292-2nd-review-package-registry-service/46917/28

Reply:
also see https://github.com/apple/swift-evolution/blob/372c605773f76da530ea7c9d20639603412a456c/proposals/0292-package-registry-service.md#security

Reply:
My [suggestion][] was to require C99 extended identifiers (for package and target names) because that's what SwiftPM uses for module names. But perhaps `XID_Start` and `XID_Continue` are similar enough to C99 extended identifiers? (I haven't investigated that yet.)

The package and target names could have the same case-sensitivity as Swift module names, unless typosquatting _within_ a scope is an issue. The scope might be normalized to lowercase? Converting between a scope (e.g. `apple`) and an identifier should also be easy, if/when prefixed module names (e.g. `import apple.NIO`) are needed.

[suggestion]: <https://forums.swift.org/t/se-0292-2nd-review-package-registry-service/46917/31>

Reply:
cc @elsh since she is exploring using package identifiers as module prefixes to address module name clashes across packages. we should confirm the constrains should largely align. imo they are ideally the same to prevent cognitive overload, and at the minimum we should confirm the constraints suggested here do not prevent us from using the package identifiers as modules names / prefixes.

Reply:
Yes, this properly addresses the concerns I had (with or without the case folding—whichever you decide).

***

For as long as it has existed, SwiftPM has successfully sanitized arbitrary names for maximum compatibility.

I looked C99 up out of curiosity, and it appears to support anything and everything beyond ASCII, but leave it up to the individual implementation whether to allow Unicode expressed directly, or only through `\U0000` escapes. That is far more lenient and flaky than UAX31, or even the Swift grammar. For example, a non‐breaking space would be allowed. I do not think C99 will be much help.